ncm-spma/src/main/pan/components/spma/functions.pan

```diff
@@ -101,8 +101,10 @@ function resolve_pkg_rep = {
     arg = repository list
 }
 function purge_rep_list = {
-    foreach (i;rep;SELF) {
-        rep['contents'] = null;
+    if (is_list(SELF)) {
```

Comment:
and how about the 
```pan
else {
   deprecated("lists don't do anything anymore");
```
or better yet use `error` (why would you use a function named `_list` on a non-list)